bitvm/src/signatures/wots_api.rs

```diff
@@ -0,0 +1,150 @@
+use paste::paste;
+use crate::treepp::Script;
+use crate::signatures::{
+    winternitz,
+    winternitz_hash::{WINTERNITZ_MESSAGE_VERIFIER, WINTERNITZ_MESSAGE_COMPACT_VERIFIER},
+};
+
+/// Trait for converting a signature into a Script.
+pub trait SignatureImpl {
+    fn to_script(self) -> Script;
+    fn to_compact_script(self) -> Script;
+}
+
+/// Macro that implements the WOTS module for a given message length (in bytes).
+/// For example:
+/// - For 160-bit WOTS, use 20 bytes.
+/// - For 256-bit WOTS, use 32 bytes.
```

Comment:
This isn't true yet, right? Currently everything seems hardcoded to 20 bytes.

Reply:
yeah, we have a macro to generate wots for arbitrary number of bytes, but the chunk/ source code has hardcoded effect of 20 bytes hash in most places. I will switch over to a variable to make it cleaner and valuable when it's required to switch over to lower hash lengths.

Reply:
cleaned up as many places inside chunk/ to make it refer to a global hash size variable, but that will not be enough to easily make the switch to a different hash size. Being able to switch to a different sized hash is sth doable and valuable. will want to defer that work in a separate PR.